File: btvn_b10/app/src/main/java/com/example/btvn_b10/MainActivity.java
package com.example.btvn_b10;

import android.os.Bundle;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class MainActivity extends AppCompatActivity {

    private RecyclerView recyclerView;
    private PostAdapter postAdapter;
    private List<Post> postList;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        recyclerView = findViewById(R.id.recyclerView);

        postList = new ArrayList<>();

        postAdapter = new PostAdapter(postList);

        recyclerView.setLayoutManager(new LinearLayoutManager(this));
        recyclerView.setAdapter(postAdapter);

        getPosts();
    }

    private void getPosts() {

        ApiService apiService = RetrofitClient.getApiService();

        Call<List<Post>> call = apiService.getPosts();

        call.enqueue(new Callback<List<Post>>() {

            @Override
            public void onResponse(
                    Call<List<Post>> call,
                    Response<List<Post>> response) {

                if (response.isSuccessful() && response.body() != null) {

                    postList.clear();
                    postList.addAll(response.body());

                    postAdapter.notifyDataSetChanged();

                } else {
                    Toast.makeText(
                            MainActivity.this,
                            "Không lấy được dữ liệu",
                            Toast.LENGTH_SHORT
                    ).show();
                }
            }

            @Override
            public void onFailure(
                    Call<List<Post>> call,
                    Throwable t) {

                Toast.makeText(
                        MainActivity.this,
                        "Lỗi: " + t.getMessage(),
                        Toast.LENGTH_LONG
                ).show();
            }
        });
    }
}

File: btvn_b6/app/src/main/java/com/example/btvn_b6/MainActivity.kt
package com.example.btvn_b6


import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView


class MainActivity : AppCompatActivity() {


    private lateinit var recyclerView: RecyclerView


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        recyclerView = findViewById(R.id.recyclerView)


        val images = listOf(
            R.drawable.img1,
            R.drawable.img2,
            R.drawable.img3,
            R.drawable.img4
        )


        recyclerView.layoutManager =
            GridLayoutManager(this,2)


        recyclerView.adapter =
            ImageAdapter(images){ image ->


                val intent =
                    Intent(this, FullImageActivity::class.java)


                intent.putExtra(
                    "image",
                    image
                )


                startActivity(intent)

            }

    }
}